tests/rptest/tests/rpk_acl_test.py

```diff
@@ -42,6 +44,43 @@ def _superclient(self):
                        password=self.superuser.password,
                        sasl_mechanism=self.mechanism)
 
+    @cluster(num_nodes=3)
+    def test_modify_then_query_error(self):
+        """
+        This test ensures that even in cases where multiple nodes may
+        think they are the leader, the freshest data is returned
+        """
+        superclient = self._superclient()
+        superclient.acl_create_allow_cluster(self.superuser.username,
+                                             "describe")
+
+        # Modify the election timeout settings
+        admin = Admin(self.redpanda)
+        admin.patch_cluster_config({'election_timeout_ms': 5000})
+
+        # Modify an ACL
+        self._rpk.sasl_allow_principal(f'User:{self.username}', ['CREATE'],
+                                       'topic', 'foo', self.superuser.username,
+                                       self.superuser.password,
+                                       self.superuser.algorithm)
+        described = superclient.acl_list()
+        assert 'CREATE' in described, "Failed to modify ACL"
+
+        # Network partition the leader away from the rest of the cluster
+        fi = make_failure_injector(self.redpanda)
```

Comment:
nit: better to use `with` construct? (to heal_all() incase something throws)

Reply:
this is done so i can pass in the node i want to isolate

Reply:
not sure what you mean, I was thinking something like..

```
with make_failure_injector(self.redpanda) as fi:
   fi.isolate(controller)
   <... do something..>

```